### PR TITLE
Fix for #33134 in v2.0

### PR DESCRIPTION
**Pull request by prabindh**
Updating softmax args description in v2.0 - logits can be of any type that can be passed to convert_to_tensor(), and Raises associated errors. #33134

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
Now that 2.0 has been released, no more changes to the branch can be accepted, except if we do security fixes and patch releases.
